src/NServiceBus.Core/MessageHandlerContextExtensions.cs

```diff
@@ -0,0 +1,19 @@
+namespace NServiceBus.Extensibility
+{
+    using System.ComponentModel;
+
+    /// <summary>
+    /// Provides extension methods for advanced MessageHandlerContext operations.
+    /// </summary>
+    public static class MessageHandlerContextExtensions
+    {
+        /// <summary>
+        /// Returns the <see cref="ContextBag"/> of the current context.
+        /// </summary>
+        [EditorBrowsable(EditorBrowsableState.Advanced)]
+        public static ContextBag GetExtensions(this IMessageHandlerContext handlerContext)
+        {
+            return ((MessageHandlerContext) handlerContext).Context;
```

Comment:
What if the context is a context bag instead of exposing one?

On Tue, Oct 20, 2015 at 3:38 PM, Tim Bussmann <notifications@github.com>
wrote:

> In src/NServiceBus.Core/MessageHandlerContextExtensions.cs
> <https://github.com/Particular/NServiceBus/pull/3022#discussion_r42495418>
> :
>
> > +namespace NServiceBus.Extensibility
> > +{
> > +    using System.ComponentModel;
> > +
> > +    /// <summary>
> > +    /// Provides extension methods for advanced MessageHandlerContext operations.
> > +    /// </summary>
> > +    public static class MessageHandlerContextExtensions
> > +    {
> > +        /// <summary>
> > +        /// Returns the <see cref="ContextBag"/> of the current context.
> > +        /// </summary>
> > +        [EditorBrowsable(EditorBrowsableState.Advanced)]
> > +        public static ContextBag GetExtensions(this IMessageHandlerContext handlerContext)
> > +        {
> > +            return ((MessageHandlerContext) handlerContext).Context;
>
> @SzymonPobiega <https://github.com/SzymonPobiega> @danielmarbach
> <https://github.com/danielmarbach> here's the first idea @andreasohlund
> <https://github.com/andreasohlund> and I came up with in order to allow
> extending the IMessageHandlerContext. It basically makes a ContextBag
> available on the context. In order to "hide" that a bit away from the user
> the first idea was to make it available via an extension method. Personally
> I start to dislike that approach because of the hard coded casting which
> makes proper unit testing impossible (MessageHandler is internal). What
> do you guys think? I'd vote for making that ContextBag available on the
> interface since that extension method doesn't really work well as a "hiding
> mechanism" as soon someone uses R# and only for hiding this to non R#
> users, I don't see enough value to justify the casting. Other ideas?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Particular/NServiceBus/pull/3022/files#r42495418>.
>



-- 
*#NSBcon 2015 • The "All About NServiceBus" conference is back *
*• Learn more <http://go.particular.net/l/27442/2015-08-21/5d4wzp> →*